packages/stitch/tests/typeMergingWithDirectives.test.ts

```diff
@@ -0,0 +1,499 @@
+    { upc: '1', inStock: true },
+    { upc: '2', inStock: false },
+    { upc: '3', inStock: true }
+  ];
+
+  const inventorySchema = makeExecutableSchema({
+    // @merge directive will direct the gateway to use the tagged resolver to resolve objects
+    // of that type. By default:
+    // 1. the key for that type will be sent to the resolvers first argument.
+    // 2. an array of keys will sent if the resolver returns a list.
+    //
+    // In this example, the key is constructed by using @base and @computed selection sets.
+    // The @computed directive for a given field instructs the gateway to only add the required
+    // additional selections when the tagged field is included within a query. In addition,
+    // the @computed directive will defer resolution of these fields even when queries originate
+    // within this subschema. The resolver for the mostStockedProduct therefore correctly returns
+    // an object with a `upc` property, but without `price` and `weight`. The gateway will use
+    // the `upc` to retrive the `price` and `weight` from the external services and return to this
+    // service for the `shippingEstimate`. Resolution for @computed fields thereby differs when
+    // querying via the gateway versus when querying the subservice directly.
+    //
+    // Note as well how the merging resolver used by the gateway includes the `price` and `weight`
+    // from the key within its internal representation of a Product even though they are not
+    // by this subschema.
+    //
+    // This example strongly types the ProductKey using an input object. This is optional,
+    // as the subschema can rely on the gateway always sending in the correctly typed data.
+    // It is typed correctly; `upc` is always specified, but `price` and `weight` will only
+    // be included when `shippingEstimate` is included within the query.
+    //
+    typeDefs: `
+      ${typeMergingDirectivesTypeDefs}
+      input ProductKey {
+        upc: String!
+        price: Int
+        weight: Int
+      }
+      type Product @base(selectionSet: "{ upc }") {
+        upc: String!
+        inStock: Boolean
+        shippingEstimate: Int @computed(selectionSet: "{ price weight }")
+      }
+      type Query {
+        mostStockedProduct: Product
+        _products(keys: [ProductKey!]!): [Product]! @merge
+      }
+    `,
+    resolvers: {
+      Product: {
+        shippingEstimate: product => {
+          if (product.price > 1000) {
+            return 0 // free for expensive items
+          }
+          return Math.round(product.weight * 0.5) || null; // estimate is based on weight
+        }
+      },
+      Query: {
+        mostStockedProduct: () => inventory.find(i => i.upc === '3'),
+        _products: (_root, { keys }) => {
+          return keys.map((key: Record<string, any>) => ({ ...key, ...inventory.find(i => i.upc === key.upc) }));
+        },
+      },
+    },
+    schemaTransforms: [typeMergingDirectivesValidator],
```

Comment:
I know this is an API of the schema package, but is this field analogous to other transforms used throughout Tools? I’ve found this to be a very confusing option, because it looks like it may do the same thing as wrap transforms with a different name.

Reply:
The schema transforms methods within wrap transforms used in wrap used to be simple functions that took the wrapping schema and gave you a new wrapping schema, but I then had to add a few arguments in version 7 so that you could add new root resolvers.

These schema transform functions are identical to the ones used with directives which are still just functions that take an executable schema and return a new executable schema